ifrs17/Import/Importers.ipynb

```diff
@@ -593,6 +571,100 @@
+                "\n        if(!dataNodes.ContainsKey(item.First().DataNode))",
+                "\n            ApplicationMessage.Log(Error.InactiveDataNodeState, item.First().DataNode);",
+                "\n}",
+                "\n"
+            ],
+            "metadata": {},
+            "execution_count": 0,
+            "outputs": []
+        },
+        {
+            "cell_type": "markdown",
+            "source": [
+                "## Validate for Data Node States Logic"
+            ],
+            "metadata": {},
+            "execution_count": 0,
+            "outputs": []
+        },
+        {
+            "cell_type": "code",
+            "source": [
+                "public async static Task ValidateDataNodeStatesAsync(this IWorkspace workspace, Dictionary<string, DataNodeData> persistentDataNodeByDataNode)",
+                "\n{",
+                "\n    foreach(var importedDataNodeState in await workspace.Query<DataNodeState>().ToArrayAsync())",
+                "\n    {",
+                "\n        if(persistentDataNodeByDataNode.TryGetValue(importedDataNodeState.DataNode, out var currentPersistentDataNode))",
+                "\n        {",
+                "\n            if(importedDataNodeState.State < currentPersistentDataNode.State)",
+                "\n                ApplicationMessage.Log(Error.ChangeDataNodeState, importedDataNodeState.DataNode, ",
+                "\n                                                                  currentPersistentDataNode.State.ToString(), ",
+                "\n                                                                  importedDataNodeState.State.ToString());",
+                "\n",
+                "\n            if(importedDataNodeState.State == currentPersistentDataNode.State)",
+                "\n                await workspace.DeleteAsync<DataNodeState>(importedDataNodeState);",
+                "\n        }",
+                "\n    }",
+                "\n}"
+            ],
+            "metadata": {},
+            "execution_count": 0,
+            "outputs": []
+        },
+        {
+            "cell_type": "markdown",
+            "source": [
+                "## Validate for Mandatory Aoc Steps"
+            ],
+            "metadata": {},
+            "execution_count": 0,
+            "outputs": []
+        },
+        {
+            "cell_type": "code",
+            "source": [
+                "public async static Task ValidateForMandatoryAocSteps(this IWorkspace workspace, IDataSet dataSet, HashSet<AocStep> mandatoryAocSteps)",
+                "\n{",
+                "\n    var excludedProperties = new[]{nameof(AocType), nameof(Novelty)};",
+                "\n    var missingAocStepsByIdentityProperties = (await workspace.Query<RawVariable>().ToListAsync())",
+                "\n                     .GroupBy(x => string.Join(\", \", x.ToIdentityString().Split(',').Where(y=> y != \" \" && !excludedProperties.Contains(y.Split(':')[0]))),",
```

Comment:
Let's allow ToIdentityString() to take inputs []params => dimension names of property we want to omit from the returned string. That should make this lambda look like: 
x => x.ToIdentityString(nameof(AocType),nameof(Novelty))

Reply:
Should I talk to devs about it?
In that case it will take I while I think.

Reply:
Did in the newest commit